MediaBrowser.Common/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,37 @@
+#nullable enable
+
+using System;
+
+namespace MediaBrowser.Common.Extensions
+{
+    /// <summary>
+    /// Extensions methods to simplify string operations.
+    /// </summary>
+    public static class StringExtensions
+    {
+        /// <summary>
+        /// Returns the part left of the <c>needle</c>.
+        /// </summary>
+        /// <param name="str">The string to seek.</param>
```

Comment:
```suggestion
        /// <param name="haystack">The string to seek.</param>
```
if you name stuff `needle` better follow the naming convention and search within a `haystack` ;)